dashboard/app/dsl/script_dsl.rb

```diff
@@ -84,26 +78,30 @@ def lesson_group(key, properties = {})
     if key
       @lesson_groups << {
         key: key,
-        display_name: properties[:display_name]
+        display_name: properties[:display_name],
+        lessons: []
       }.compact
     end
-    @lesson_group = key
   end
 
   def lesson(name, properties = {})
-    if @lesson
-      @lessons << {
-        lesson: @lesson,
-        visible_after: @lesson_visible_after,
-        scriptlevels: @scriptlevels,
+    # For scripts that don't use lesson groups create a blank non-user facing lesson group
+    if name
+      if @lesson_groups.empty?
```

Comment:
I thought we already had logic for creating a default lesson group in `add_script`. Will it be duplicated now?

Reply:
This just makes sure we create a lesson group to hold the lessons in the DSL output if one is not specified by the curriculum writer.